api/request_source/service.proto

```diff
@@ -37,9 +38,18 @@ message RequestSpecifier {
   // Request content length. The client will transfer the number of bytes specified here for the
   // request body.
   google.protobuf.UInt32Value content_length = 4;
-  // Request header replacements. Any existing header(s) with the same name will be removed
-  // before setting.
-  envoy.api.v2.core.HeaderMap headers = 5;
+
+  oneof oneof_headers {
```

Comment:
Comparing `HeaderMap` [v2](https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/core/base.proto) versus [v3](https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/core/v3/base.proto), it seems that there aren't any actual changes to the message types that we depend upon, definitions seem identical to me.

That leads me to think we might be able to get away with a mechanical on the fly conversion if needed? The v3 messages have a `previous_message_type` attached in their definition, can we access that at runtime?

The on the fly v2/v3 conversion will not be free, but for the gRPC case I suspect that it won't bottleneck, so maybe this is worth considering.

Reply:
As discussed offline, this was on my radar when working on this PR, but considering the added complexity and deps for the reflection that would load and convert this message mechanically and the size of the message in question (just two fields), it didn't seem like the mechanical solution would pay for itself.

With that we are choosing to keep this as is since the current code is simple to understand and maintain.